wrapper/pre_copy.py

```diff
@@ -716,7 +763,8 @@ def _start_nbdkits(self, final):
                 filters.remove(filt)
 
         for disk in self.disks:
-            cmd = self._get_nbdkit_cmd(disk, filters, final)
+            cmd = getattr(self, "_get_nbdkit_cmd_%s" % self._transport_method)(
+                    disk, filters, final)
```

Comment:
let's say `self._transport_method` contains something invalid where will the `AttributeError` generated by `getattr()` be handled?

Reply:
`transport_method` value is checked here: https://github.com/ManageIQ/manageiq-v2v-conversion_host/blob/master/wrapper/virt_v2v_wrapper.py#L392. So at this stage it will already have been caught.

Reply:
I am not saying this needs fixing (just yet), but the check is in a separate part of code. Assuming we change the check or behavior later to support more protocols and we forget to add a new method here, where will the error be caught?

Reply:
Validating the transport methods should be done in the `validate_data` method for the provider. Currently, there's only once source provider, VMware, so it's still done in `virt-v2v-wrapper.py`. IMO, the VMware provider would deserve its own `VmwareHost` class. But the `hosts.py` becomes extra large, so we should work to split the code, maybe with more interfaces/factories.